frontend/app/routes/protected/person-case/current-name.tsx

```diff
@@ -129,27 +146,37 @@ export async function action({ context, request }: Route.ActionArgs) {
             }),
           ],
           t('protected:current-name.supporting-error.required-error'),
-        ),
-      ]) satisfies v.GenericSchema<CurrentNameSessionData>;
+        );
 
-      const input = {
-        firstName: String(formData.get('first-name')),
-        middleName: String(formData.get('middle-name')),
-        lastName: String(formData.get('last-name')),
-        supportingDocumentsRequired: formData.get('docs-required')
-          ? formData.get('docs-required') === REQUIRE_OPTIONS.yes
-          : undefined,
-        supportingDocumentTypes: formData.getAll('doc-type') as string[],
-      } satisfies Partial<v.InferInput<typeof schema>>;
+        const schema = v.intersect([
+          differentNameSchema,
+          supportingDocsSchema,
+        ]) satisfies v.GenericSchema<CurrentNameSessionData>;
 
-      const parseResult = v.safeParse(schema, input, { lang });
+        const input = {
+          preferredSameAsDocumentName: formData.get('same-name')
+            ? formData.get('same-name') === REQUIRE_OPTIONS.yes
+            : undefined,
+          firstName: String(formData.get('first-name')),
+          middleName: optionalString(formData.get('middle-name') as string),
+          lastName: String(formData.get('last-name')),
+          supportingDocumentsRequired: formData.get('docs-required')
+            ? formData.get('docs-required') === REQUIRE_OPTIONS.yes
+            : undefined,
```

Comment:
Why set this to `undefined` instead of a simple true/false?
ex:
``` ts
supportingDocumentsRequired: formData.get('docs-required') === REQUIRE_OPTIONS.yes
```